### DISCUSSION TITLE
How to make a token dispenser

### DISCUSSION THREAD

**Discussion by ayesparshh** [Cyfrin/foundry-full-course-cu#2293] [General]
Like in all the class and assignment completion we get a good gesture token how to implement it ? 
I want to implement it for a course selling company for every milestone they will get a token as a completion gesture.

**Comment by EngrPips** [comment]
Hey @ayesparshh, I guess it's time to do some research

**Reply by ayesparshh** [reply]
I'm in a hurry that's why I asked here to get the on point approach to implement this.

**Reply by EngrPips** [reply]
You can't be in a hurry if you want to implement a codebase. Take your time and do the necessary research. You can look at the `Foundry-Full-Course` NFT contract for inspiration.

**Reply by ayesparshh** [reply]
I've been searching for the code related to this contract but haven't been able to find it. Could you please share the link where this contract is written?
Thanks.

**Reply by EngrPips** [reply]
You can visit this link https://sepolia.etherscan.io/address/0x76B50696B8EFFCA6Ee6Da7F6471110F334536321#readContract to get the `challenge contract` of any section, and you should take time to understand the flow of the contract and how it will call the `NFT` contract to mint

**Reply by ayesparshh** [reply]
Thank you so much.

**Reply by EngrPips** [reply]
You are welcome, Friend.
